Updated with killer info

Hodareh opened the pull request:
Turtle now displays who killed you. Updated the addon to include this new info.

Hodareh commented:
Hi! I wrote this some months ago and finally got around to putting it in a repo so I can PR it to you.

zeress commented:
Thanks for the update Hodareh; would it be possible to also add the death reason as a log entry?

Hodareh commented:
Hi zeress,

I've added the killer info to the log. Unfortunately, the addon will error attempting to display the log frame if any of the log entries do not have killer info. A user could manually edit the addon's SavedVariables file to fix it (if they want to keep the extant log entries), or they can purge the log. Since the purge button was on the log frame, I've added a new command to provide an alternative method, "/hcd purge".

trumpetx commented:
@zeress & @Hodareh  - see my PR to fix the nil problem (keep your old data!)

https://github.com/Hodareh/HardcoreDeath/pulls
